resources/providers/gcplib/auth/credentials.go

```diff
@@ -0,0 +1,41 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gcplib
+
+import (
+	"errors"
+	"github.com/elastic/cloudbeat/config"
+	"google.golang.org/api/option"
+)
+
+func GetGcpClientConfig(cfg *config.Config) ([]option.ClientOption, error) {
+	// Create credentials options
+	gcpCred := cfg.CloudConfig.GcpCfg
+	var opt []option.ClientOption
+	if gcpCred.CredentialsFilePath != "" && gcpCred.CredentialsJSON != "" {
+		return nil, errors.New("both credentials_file_path and credentials_json specified, you must use only one of them")
+	} else if gcpCred.CredentialsFilePath != "" {
```

Comment:
could we favor one instead of returning an error?

this would play nicer with the way our integration installation/update works, as the form will send all fields filled by the user, so in the case they do fill both fields, they will end up learning it didn't work from the logs. 

maybe even even try both if one of them happens to not work. 

wdyt?

Reply:
@orouz - I was wondering about this.
Do you think we should add a verification to the user input? 
For example, in case the SA file path does not exist, fallback to the other option? or leave this responsibility to the user.
Same as for the JSON content.

Reply:
thinking on this again, maybe we should just favor the JSON? seems like the file path would be used for the actual prod env and the json would be for development, so preferring it would allow a better DX. 

i'm good either way, just figured since we do have values, best to use them